components/layout_2020/flexbox/layout.rs

```diff
@@ -299,16 +300,37 @@ impl FlexContainer {
             },
         };
 
+        let row_gap = container_style.clone_row_gap();
+        let column_gap = container_style.clone_column_gap();
+        let (cross_gap, main_gap) = match flex_context.flex_axis {
+            FlexAxis::Row => (row_gap, column_gap),
+            FlexAxis::Column => (column_gap, row_gap),
+        };
+        let cross_gap = match cross_gap {
+            LengthPercentageOrNormal::LengthPercentage(length_percent) => length_percent
+                .maybe_to_used_value(flex_context.container_definite_inner_size.cross)
+                .unwrap_or_default(),
+            LengthPercentageOrNormal::Normal => Au::zero(),
+        };
+        let main_gap = match main_gap {
+            LengthPercentageOrNormal::LengthPercentage(length_percent) => length_percent
+                .maybe_to_used_value(flex_context.container_definite_inner_size.main)
+                .unwrap_or_default(),
+            LengthPercentageOrNormal::Normal => Au::zero(),
+        };
+
         // “Resolve the flexible lengths of all the flex items to find their *used main size*.”
         // https://drafts.csswg.org/css-flexbox/#algo-flex
         let flex_lines = collect_flex_lines(
             &mut flex_context,
             container_main_size,
             &mut flex_items,
-            |flex_context, mut line| line.layout(flex_context, container_main_size),
+            main_gap,
         );
 
-        let content_cross_size = flex_lines.iter().map(|line| line.cross_size).sum();
+        let line_count = flex_lines.len();
+        let content_cross_size = flex_lines.iter().map(|line| line.cross_size).sum::<Au>() +
+            cross_gap * (line_count as i32 - 1);
```

Comment:
Have you tested for possible divide-by-zero here? (perhaps servo guarantees at least one line but that isn't obvious from the diff).

Reply:
`collect_flex_lines` returns a vec with at least one line...